fetchai/ledger/serialisation/transaction.py

```diff
@@ -278,11 +300,21 @@ def decode_transaction(stream: io.BytesIO) -> (bool, transaction.Transaction):
     # extract all the signing public keys from the stream
     public_keys = [identity.decode(stream) for _ in range(num_signatures)]
 
+    for ident in public_keys:
+        tx._signers[ident] = {}
+
+    return tx
+
```

Comment:
Here we have split out one method into two, and that is fine, but the javascript has different way of doing things in this place since we don't have tellall on a buffer so things are different and cannot so cleanly replicate it. It seems from what I can see to be cosmetic and since I have not done this in javascript, and maybe we can revert this small refactor in the python if it does nothing as i believe